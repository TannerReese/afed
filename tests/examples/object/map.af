# Test map operations

# Indexing
mp: {1, 2, x: {"###": "$abc$"}, y: 100, 3, 4},
mp "x" "###" = ``,
mp 3 = ``,
mp "hello" = ``,

# Merge maps
new: {1, 2, a: "b"} + {6, "c": "d"},
new    "a" = ``,
new  .  c = ``,

# Empty map
{} = ``,

# Keys cannot be integers
{3: 4} = ``,

# Map unimplemented operators
{a: 4} * {b: 6} = ``,
5 / {r: 3} = ``,

