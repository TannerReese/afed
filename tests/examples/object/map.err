Parse Error in line 13, column 3
	{3: 4} = ``,
	  ^
Error: Extra content in member

1 Parse Error encountered


line 6, column 8 Eval Error: Expected string, but found number
line 7, column 14 Eval Error: Key hello is not contained in map
line 16, column 36 Eval Error: Binary operator + not implemented between types map and map
line 17, column 19 Eval Error: Binary operator * not implemented between types map and map
line 18, column 14 Eval Error: Binary operator / not implemented between types number and map
5 Eval Errors encountered
