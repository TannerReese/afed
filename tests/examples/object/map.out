# Test map operations

# Indexing
mp: {1, 2, x: {"###": "$abc$"}, y: 100, 3, 4},
mp "x" "###" = `"$abc$"`,
mp 3 = `Eval Error: Expected string, but found number`,
mp "hello" = `Eval Error: Key hello is not contained in map`,

# Empty map
{} = `{}`,

# Keys cannot be integers
{3: 4} = `{}`,

# Map unimplemented operators
{1, 2, "a": "b"} + {6, "c": "d"} = `Eval Error: Binary operator + not implemented between types map and map`,
{a: 4} * {b: 6} = `Eval Error: Binary operator * not implemented between types map and map`,
5 / {r: 3} = `Eval Error: Binary operator / not implemented between types number and map`,

